packages/components/src/font-size-picker/types.ts

```diff
@@ -81,18 +76,25 @@ export type FontSize = {
 	slug: string;
 };
 
-export type FontSizeOption = Omit< FontSize, 'size' > &
-	Partial< Pick< FontSize, 'size' > >;
-
-export type FontSizeSelectOption = Pick< FontSizeOption, 'size' > & {
-	key: string;
-	name?: string;
-	__experimentalHint: ReactNode;
+export type FontSizePickerSelectProps = {
+	fontSizes: FontSize[];
+	value: FontSizePickerProps[ 'value' ];
+	size: FontSizePickerProps[ 'size' ];
+	onChange: ( value: number | string | undefined ) => void;
+	onSelectCustom: () => void;
 };
```

Comment:
We can reuse more prop types from `FontSizePickerProps` by using the `NonNullable` utility, and we can use `Pick` for more clarity:

```suggestion
export type FontSizePickerSelectProps = Pick<
	FontSizePickerProps,
	'value' | 'size'
> & {
	fontSizes: NonNullable< FontSizePickerProps[ 'fontSizes' ] >;
	onChange: NonNullable< FontSizePickerProps[ 'onChange' ] >;
	onSelectCustom: () => void;
};
```